database/sqlite3-driver/lite3_database.ml

```diff
@@ -81,14 +82,39 @@ end
 
 class lite3_database file = object(self)
   inherit Value.database
+  val mutable _supports_shredding : bool option = None
   val connection = db_open file
   method exec query : Value.dbvalue =
+    Debug.print query;
     let stmt = prepare connection query in
       new lite3_result stmt
   (* See http://www.sqlite.org/lang_expr.html *)
   method escape_string = Str.global_replace (Str.regexp_string "'") "''"
-  method quote_field = self#escape_string
+  method quote_field s = "\"" ^ self#escape_string s ^ "\""
   method driver_name () = "sqlite3"
+  method supports_shredding () =
+    match _supports_shredding with
+    | Some result -> result
+    | None ->
+       let stmt = prepare connection "SELECT sqlite_version()" in
+       let result = new lite3_result stmt in
+       match result#status with
+       | `QueryOk ->
+          let version = result#getvalue 0 0 in
+          let is_number str =
+            Str.string_match (Str.regexp "^[0-9]+$") str 0
+          in
+          begin match Str.split (Str.regexp "[0-9]\\.") version with
+          | major :: minor :: _ when is_number major && is_number minor ->
```

Comment:
This looks fine, but given that different databases have their own idiosyncratic way of reporting versions, it may be a pain to maintain.  If so I'm also happy with just having a hardwired true/false constant per driver and we just document the needed versions.

Reply:
Indeed, that is I have opted for the conservative option. If the version number is not recognisable, then the driver will report that it does not support shredding. I suppose, we can improve this in the future.